test/functorch/test_vmap.py

```diff
@@ -3900,6 +3900,35 @@ def test_vmap_multi_dot_failure_1D_input(self):
         with self.assertRaisesRegex(RuntimeError, "tensor 1 must be 2D but got 1D"):
             return vmap(torch.linalg.multi_dot)(inputs)
 
+    def test_vmap_escaped_error(self):
```

Comment:
I would suggest doing both `assertRaises` in same test as they share most of the code.

```python
def test_vmap_escaped_error(self):
    escaped = None

    def f(x):
        nonlocal escaped
        escaped = x
        return x ** 2

    x = torch.randn(3)
    vmap(f)(x)

    with self.assertRaisesRegex(RuntimeError, "your  tensor escaped from vmap"):
        escaped.sin()
    
    with self.assertRaisesRegex(RuntimeError, "your boxed tensor escaped from vmap"):
        escaped.sin_()

```

Reply:
fixed